exchanges/bitstamp/bitstamp_type_convert.go

```diff
@@ -0,0 +1,29 @@
+package bitstamp
+
+import (
+	"encoding/json"
+	"regexp"
+	"strconv"
+)
+
+var currRE = regexp.MustCompile(`^[\d.]+`)
+
+// UnmarshalJSON deserializes JSON, and timestamp information.
+func (p *TradingPair) UnmarshalJSON(data []byte) error {
+	type Alias TradingPair
+	t := &struct {
+		*Alias
+		MinimumOrder string `json:"minimum_order"`
+	}{
+		Alias: (*Alias)(p),
+	}
+
+	err := json.Unmarshal(data, t)
+	if err == nil {
+		if m := currRE.FindString(t.MinimumOrder); len(m) > 0 {
```

Comment:
This is slightly more performant than using regular expression:
```diff
diff --git a/exchanges/bitstamp/bitstamp_type_convert.go b/exchanges/bitstamp/bitstamp_type_convert.go
index abb72525a..959fa670f 100644
--- a/exchanges/bitstamp/bitstamp_type_convert.go
+++ b/exchanges/bitstamp/bitstamp_type_convert.go
@@ -2,12 +2,11 @@ package bitstamp

 import (
        "encoding/json"
-       "regexp"
+       "fmt"
        "strconv"
+       "strings"
 )

-var currRE = regexp.MustCompile(`^[\d.]+`)
-
 // UnmarshalJSON deserializes JSON, and timestamp information.
 func (p *TradingPair) UnmarshalJSON(data []byte) error {
        type Alias TradingPair
@@ -19,11 +18,15 @@ func (p *TradingPair) UnmarshalJSON(data []byte) error {
        }

        err := json.Unmarshal(data, t)
-       if err == nil {
-               if m := currRE.FindString(t.MinimumOrder); len(m) > 0 {
-                       p.MinimumOrder, err = strconv.ParseFloat(m, 64)
-               }
+       if err != nil {
+               return err
+       }
+
+       before, _, found := strings.Cut(t.MinimumOrder, " ")
+       if !found {
+               return fmt.Errorf("unhandled minimum order string: %s", t.MinimumOrder)
        }

+       p.MinimumOrder, err = strconv.ParseFloat(before, 64)
        return err
 }
```

Reply:
That is done

Reply:
The currency suffix isn't documented, ergo could be subject to change.
If we use Cut like this then I think we should handle t.MinimumOrder as a fallback when not found.